msteams-platform/concepts/messaging-extensions/create-extensions.md

```diff
@@ -339,22 +352,130 @@ This used to insert a card into the compose box as a result of a the command. It
               ]
             }
           ]
-        },
-        "preview": {
-          "contentType": "application/vnd.microsoft.card.thumbnail",
-          "content": {
-            "title": "85069: Create a cool app",
-            "images": [
+        }
+      }
+    ]
+  }
+}
+```
+
+### Respond with an adaptive card message sent from a bot
+
+> [!NOTE]
+> Responding with an adaptive card message sent from a bot is in [developer preview](~/resources/dev-preview/developer-preview-intro.md). Currently the only type of response attachment supported is an adaptive card.
+
+You can also respond to the submit action by inserting a message with an [adaptive card](~/concepts/cards/cards.md#adaptive-cards) into the channel with a bot. Your user will be able to preview the message before submitting it, and potentially edit/interact with it as well. This can be very useful in scenarios where you need to gather information from your users before creating an adaptive card response. The following scenario shows how you can use this flow to configure a poll without including the configuration steps in the channel message.
+
+1. The user clicks the messaging extension to trigger the task module.
+1. The user uses the task module to configure the poll.
+1. After submitting the configuration task module the app uses the information provided in the task module to craft an adaptive card and sends it as a `botMessagePreview` response to the client.
+1. The user can then preview the adaptive card message before the bot will inserts it into the channel. If the bot is not already a member of the channel, clicking `Send` will add the bot.
+1. Interacting with the adaptive card will change the message before sending it.
+1. Once the user clicks `Send` the bot will post the message to the channel.
+
+To enable this flow your task module should respond as in the example below, which will present the preview message to the user.
+
+```json
+{
+  "composeExtension": {
+    "type": "botMessagePreview",
+    "activityPreview": {
+      "attachments":  [
+        {
+          "contentType": "application/vnd.microsoft.card.adaptive",
+          "content": << Card Payload >>
+        }
+      ]
+    }
+  }
+}
+```
+
+Your message extension will now need to respond to two new types of interactions, `value.botMessagePreviewAction = "send"` and `value.botMessagePreviewAction = "edit"`. Below is an example of the `value` object you will need to process:
+
+```json
+{
+  "name": "composeExtension/submitAction",
+  "type": "invoke",
+  ...
+  "value": {
+    "botMessagePreviewAction": "send",
+    "botActivityPreview": [
```

Comment:
The posting back payload should be simply "echo back" the Activity,  so follows the Activity schema (see above). However, due to SMBA message splitting, instead of single-out payload, we'll have multiple-in echo back. `botActivityPreview` is therefore an array of Activity. An example: 

```
    "value": {
        "botMessagePreviewAction": "edit" | "send",
        "botActivityPreview": [
            {
                "type": "message/card",
                "attachments": [
                    {
                        "content": {},
                        "contentType": "application/vnd.microsoft.card.adaptive"
                    }
                ]
            }
        ],
        "context": {
            "theme": "default"
        }
    }
```

These are "ready-to-send" payloads @clearab FYI

Reply:
@robin-liao Why does message splitting enter the picture, if `activityPreview` **must** be an activity w/ 1 adaptive card?
And why the name change from activityPreview -> botActivityPreview in this case?